### ISSUE TITLE
Fix concerning quotation marks

### ISSUE THREAD

**Issue by vivadavid** [SubtitleEdit/subtitleedit#318]
Hello,

I've been posting some comments on Nikse's blog, as well as via e-mail, but this is the first time I write here at the forum. I've known SE for quite some time now, but I recently decided to use it regularly to, among other things, perform OCR on PSG files.

As far as I know, automatic fixes come from a file called spa_OCRFixReplaceList.xml and I have a number of comments about these automatic corrections. I think it'll be easier if I create a different thread for each one of these.

My number 1 problem so far with SE, when performing OCR, is that it stops every time there's a word starting with `«` or `?` (we have in Spanish an opening exclamation mark) or ending with `»`. SE wouldn't recognize words such as (this example is random):

```
«coche
?coche
coche»
```

even if the word "coche" is perfectly common and is on the dictionary. This happens too often during the OCR process and it's really frustrating because those are, most of the time, correct words and I have to systematically click on Ignore All.

One way to fix this would be by just ignoring those punctuation signs, but the problem, in my opinion, is deeper. At some point, somebody must have suggested to make the following replacements:

`"` -> `«`
`"` -> `»`

(Let's include here all types of double quotation marks apart from the straight ones, such as “)

I suggest using straight double quotation marks (") in all situations, avoiding « and » (and all types of quotation marks with a different opening and closing sign) for the following reasons:
1. First of all, because of the reason I stated earlier: SE believes words with those punctuation signs are different words.
2. I've found many examples where SE asks about a word starting with `«` and then does nothing with the closing quotation mark. The result is something like this:
   
   `«El lugar esta bendito".`
3. I can't think of a time where I have seen the quotation marks `«` and `»` in Spanish subtitles. I always see the double quotation marks.
4. Straight quotation marks (`"`) are simpler than the other type (`“`).
5. You already find those straight double quotation marks in most subtitles, so there'd be nothing to change or replace.
6. Using a sign that is the same to open and close a quotation avoids the problem of finding out if we're opening or closing a quotation.

To sum up, my suggestions are:
1. To avoid `«` and `»`.
2. To use straight double quotation marks (the same sign to open and close).
3. To ignore the opening exclamation mark (?) when checking a word, though SE should still look at
   the word without the exclamation mark, just in case it's wrong.

I hope my first thread wasn't too long!

Thanks and, as I've told you before, congratulations on this great software, @niksedk!

**Comment by vivadavid** [comment]
Hello, Nikse,

I don't want to seem pushy, but I reported this issue last October and nobody has said anything about it. I think it's an important issue and I hope it's dealt with in the foreseeable future.

Please let me know if I didn't explain the whole thing very well.

Regards,

David

**Comment by orbittwz** [comment]
Hey
It happened to me also, the first OCR file went fine, and in the second, every line has "?" in the beginning of it.
Can someone fix that?

**Comment by niksedk** [comment]
@orbittwz: Could you explain exactly what happens and what you expected + sample file?

**Comment by orbittwz** [comment]
OK, in the first scenario when it happens, i dropped files to batch OCR, checked "remove text for HI", "multiple replace" and "fix RTL"...
the first file went fine, see sample1.srt
the second file got added with weird ? on EVERY line in the beginning of it.

[sample.zip](https://github.com/SubtitleEdit/subtitleedit/files/2896854/sample.zip)

Though i have an update, those files were made in Tesseract4 , but now i used Tesseract302... and didn't checked "multiple replace" and "fix RTL", and now the OCRing went fine.
The main thing i suspect is the FIX RTL, seems the options within it are messing the lines.

**Comment by orbittwz** [comment]
and if i got both Tesseracts, it will defaults to 302?
![image](https://user-images.githubusercontent.com/45605734/53287320-315e0700-3783-11e9-9242-7a21cc89435d.png)

**Comment by OmrSi** [comment]
Yeah, did you change the encoding from UTF-8 to ANSI?

**Comment by orbittwz** [comment]
I changed it to my language encoding, which is Hebrew MAC 10005

**Comment by OmrSi** [comment]
I'm guessing you did, you used the method "Fix RTL via Unicode Control character" and the file had an encoding other than UTF-8 or it was UTF-8 and you changed it to ANSI.
This method adds an invisible Unicode character, the keyword here is "Unicode" so it will only work with UTF, when you change it to another encoding the character isn't recognized and appears as a "?"
The solution is to use the other reversing method which is "Reverse RTL start\end" or leave the file in UTF-8.

**Comment by orbittwz** [comment]
Yeah, it probably the solution here, the fit RTL checked is the problem.
So for me, the best way is to first OCR without that, then translating to my language, and then using the other batch convert options on those files which are in my language...
but the very important lesson from this is backup backup backup. :)

**Comment by orbittwz** [comment]
yep, seem like the same issue...
so now both cases can be closed
thank you!

**Comment by OmrSi** [comment]
See this for more info https://github.com/SubtitleEdit/subtitleedit/issues/3316

**Comment by OmrSi** [comment]
@orbittwz Yeah, backup always is a lifesaver. :D
Yeah, you should translate then reverse using the other method if you want to save the files in a different encoding. 👌👌
You got it!

**Comment by orbittwz** [comment]
💯
